### PR TITLE
Fixes for listing overflow, and bad input when creating a listing

### PR DESCRIPTION
**Pull request by MichaelFauht**
**1. Listing Overflow**

Listings will now render in rows of 3 listings, with new rows beginning when the amount of listings displayed exceeds 3

image

**2. Empty Input**

If a user does not enter all the required fields to create a listing, they will be notified with a brief pop up

image

**3. Price**

Users may enter the price in a variety of formats such as $ 30, 30 $, or 30. A proper float will be extracted from their input with regex

### CODE REVIEW

File: pages/home.js [hunk 1]
```diff
@@ -3,101 +3,101 @@ import { Container } from "@mui/system";
...
-        <Container>
-            <ResponsiveAppBar />
-            <div className="titleHeader">
-                <h1 className="title">Your Saved Listings</h1>
-                {renderListings()}
-            </div>
-        </Container>
+      <div class="col-md-4">
+        <SavedListing
+          key={i}
```
Comment (axelnguyen0701) [inline pages/home.js]: Hey I think you forgot to move the key props to the outside div. This way react can keep track of what to render next.

File: pages/createlisting.js [hunk 2]
```diff
@@ -11,401 +11,420 @@ import Button from "@mui/material/Button";
...
+              component="span"
+            >
+              Upload Image
+            </Button>
+          </ThemeProvider>
+        </div>
+      </label>
+      {imageUrl && selectedImage && (
+        <div className="search-button-box" mt={2} textAlign="center">
+          <img src={imageUrl} alt={selectedImage.name} height="300px" />
```
Comment (axelnguyen0701) [inline pages/createlisting.js]: From NextJS documentation: "Using `` could result in slower LCP and higher bandwidth. Use `<Image />` from `next/image` instead to utilize Image Optimization. See: https://nextjs.org/docs/messages/no-img-element"

File: pages/home.js [hunk 3]
```diff
@@ -3,101 +3,101 @@ import { Container } from "@mui/system";
...
+  const renderedListings = nfts.map((e, i) => {
     return (
-        <Container>
-            <ResponsiveAppBar />
-            <div className="titleHeader">
-                <h1 className="title">Your Saved Listings</h1>
-                {renderListings()}
-            </div>
-        </Container>
+      <div class="col-md-4">
```
Comment (axelnguyen0701) [inline pages/home.js]: Change props `class` to `className`

File: pages/home.js [hunk 4]
```diff
@@ -3,101 +3,101 @@ import { Container } from "@mui/system";
...
+
+    return renderedListings;
+  };
+
+  return (
+    <Container>
+      <ResponsiveAppBar />
+      <div className="titleHeader">
+        <h1 className="title">Your Saved Listings</h1>
+        <div class="row"> {renderListings()}</div>
```
Comment (axelnguyen0701) [inline pages/home.js]: Change props class to className

**Review by axelnguyen0701 — CHANGES_REQUESTED** [review]
Looks good to me, just need to change some small details

File: pages/createlisting.js (hunk 2 above)
Comment (MichaelFauht) [reply pages/createlisting.js]: Interesting, I didn't know that. That's good to know

**Comment by MichaelFauht** [comment]
I've made the requested changes

**Review by axelnguyen0701 — APPROVED** [review]

**Comment by axelnguyen0701** [comment]
Sorry for the delay! Good work!

**Pull request merged**